### ISSUE TITLE
Match Properties bug/unexpected behaviour

### ISSUE THREAD

**Issue by djdarkstream** [eirannejad/pyRevit#859]
**Describe the bug**
I was matching a window and did check all, but the window did not match the properties correctly. (only height was changed to the right value) 
I then selected sill height, height and head-height because these values were the ones that had to change, but this did also not have the wanted effect (exactly same as check all).

If I change only the sill height and height properties, it works (the end result is as expected), this might be because the head height is calculated with the sill- and height properties. At least that's my first thought.

If at all possible it would be nice if the system recognizes I want to change a windows sill- and head- & height and that it would omit one of these so the calculation is done correctly. If it does that then I won't need to specifically select sill- and height and that would allow me to use the check all button.

I think it could be done by checking the request of the user and if sill height and/or head height is a property that has to change, then omit one of these values (not sure what would be wise but I guess that you could answer that better.

**To Reproduce**
1. Go to Architecture
2. Click on Window and place two of them in a wall
3. Change the sill- & height of one of the two and then match properties on the values that are unequal or just use check all
4. The window that was to match the other, now is at the default sill height with the proper height parameter. See screenshot

**Expected behavior**
Fully matching the properties of the other when selecting sill-, head- & height / check all

**Screenshots**
![match prop](https://user-images.githubusercontent.com/5404802/77168717-97b57600-6ab8-11ea-9c1b-0e2d4e234b0f.png)
![match prop result](https://user-images.githubusercontent.com/5404802/77168722-9a17d000-6ab8-11ea-887f-e6d82f0425e6.png)

**Desktop (please complete the following information):**
 - OS: Win10 Pro 1909
 - pyRevit Version 4.7.5:4bdb321
 - pyRevit Environment: 

```
==> Registered Clones (full git repos)
master | Branch: "master" | Version: "4.7.5:4bdb321" | Path: "C:\Users\Administrator\AppData\Roaming\pyRevit-Master"
==> Registered Clones (deployed from archive/image)
==> Attachments
master | Product: "2019.2.2 Hotfix" | Engine: 277 | Path: "C:\Users\Administrator\AppData\Roaming\pyRevit-Master" | Manifest: "C:\Users\Administrator\AppData\Roaming\Autodesk\Revit\Addins\2019\pyRevit.addin"
==> Installed Extensions
==> Default Extension Search Path
C:\Users\Administrator\AppData\Roaming\pyRevit\Extensions
==> Extension Search Paths
==> Extension Sources - Default
https://github.com/eirannejad/pyRevit/raw/master/extensions/extensions.json
==> Extension Sources - Additional
==> Installed Revits
2019.2.2 Hotfix | Version: 19.2.20.24 | Build: 20190808_0900(x64) | Language: 1033 | Path: "C:\Program Files\Autodesk\Revit 2019"
==> Running Revit Instances
PID: 244 | 2019.2.2 Hotfix | Version: 19.2.20.24 | Build: 20190808_0900(x64) | Language: 0 | Path: "C:\Program Files\Autodesk\Revit 2019"
==> User Environment
Microsoft Windows 10 [Version 10.0.18363]
Executing User: HP-LAPTOP-DOUG\Administrator
Active User: HP-LAPTOP-DOUG\Administrator
Admin Access: No
%APPDATA%: "C:\Users\Administrator\AppData\Roaming"
Latest Installed .Net Framework: 4.8
No .Net Target Packs are installed.
No .Ne-Core Target Packs are installed.
pyRevit CLI 0.21.0.0
```

**Additional context**
I noticed when I did the check all that the markdown was also matched, in most use cases I would expect that one would not want to have that changed to match the other seeing all they should all be unique. Might be an idea to remove that property from the check all option or have another workaround if it sometimes would make sense to match it.

**Comment by djdarkstream** [comment]
I just did another Match properties with only height and head height but this resulted in a window that was the same size, but the head height was unchanged. Thought I'd add it because it seems that there is more going on then I first expected.

**Comment by eirannejad** [comment]
Great issue report. Thanks a lot. Let me take a look at the condition and will keep you posted.

**Comment by eirannejad** [comment]
This seems to be a window family issue. Revit windows have builtin Width, Height that are both Type parameters. I'm not sure why these windows have Width, Height instance parameters but there might be conflicts there. I modified the tool to apply the type properties first, and then move on to the instance properties. This avoid the conflict between Revit updating the "Head Height" automatically when the window size changes, and pyRevit attempting to update the same value

![bw8hH0RmdB](https://user-images.githubusercontent.com/8197916/90792176-047aa980-e2bf-11ea-9f8a-c1ec30289dd5.gif)

Fixed and will be published with next release. Closing this for now. Test the tool when the new version is published and feel free to open this again if the issue persists.
